api/client.js

```diff
@@ -102,39 +108,77 @@ class Client {
     // Instantiate the transport, used for sending messages to the server.
     this._transport = new Transport(ws, options.closeCallback, options.debug);
     this._facades = options.facades;
-    this._adminFacadeVersion = options.adminFacadeVersion;
+    this._bakery = options.bakery;
+    this._admin = new Admin(this._transport, {});
   }
 
   /**
     Log in to Juju.
 
     @param {Object} credentials An object with the user and password fields for
       userpass authentication or the macaroons field for bakery authentication.
+      If an empty object is provided a full bakery discharge will be attempted
+      for logging in with macaroons. If the provided macaroons need discharge,
+      or no macaroons neither userpass are provided, the bakery instance
```

Comment:
I'm not really sure what this sentence is saying...

Reply:
Rephrased.